javascript/javascript2/week2/hyfBay/main.js

```diff
@@ -0,0 +1,30 @@
+console.log("Script loaded");
+
+const products = getAvailableProducts();
+
+const filter = document.getElementById("filter");
+let h3 = document.getElementsByName("h3");
+filter.addEventListener("keyup", (e) => {
+  const srarchh = e.target.value;
+  const fil = h3.filter((p) => {
```

Comment:
here there is an error because you are trying to run an array method on HTML element. so Array.from(h3).filter(...) would solve that and you could be then checking for p.innerText to see whether it includes your search word. Also, it is good to make the search case insensitive by converting the product title and search input into .toLowerCase

However, it would not bring you to what you probably actually want to do - show only products that fit the search input . To achieve that you would want to filter through products and check their name and then render those by using your great renderProducts() function 

```
const pUl = document.createElement("ul"); // moving this one before the filter function so one has access to the list 
pUl.classList.add("products-list");
document.querySelector("main").appendChild(pUl);

filter.addEventListener("keyup", (e) => {
  pUl.innerHTML = ""; // clears the page so it doesnt show all the other products
  const srarchh = e.target.value.toLowerCase();
  const fil = products.filter((p) => {
    return p.name.toLowerCase().includes(srarchh);
  });
  renderProducts(fil) // you also want to call your render function to show the products
});
```